docs/docs/script/caching.py

```diff
@@ -0,0 +1,343 @@
+# 
+# The `CloudPath.open` method supports a `force_overwrite_from_cloud` kwarg to force overwriting your local version.
+# 
+# `OverwriteNewerCloudError`
+# This exception is raised if we are asked to upload a file, but the one on the cloud is newer than our local version. This likely means that a separate process has updated the cloud version, and we don't want to overwrite and lose that new data in the cloud.
+# 
+# The `CloudPath.open` method supports a `force_overwrite_to_cloud` kwarg to force overwriting the cloud version.
+# 
+# 
+
+# # Clearing the file cache
+# 
+# There's no perfect strategy for when to clear the file cache, and different applications will have different requirements and preferences. `cloudpathlib` provides fine-grained control over when cloud files are removed from the local disk. The cache can be emptied both manually and automatically. Because `cloudpathlib` uploads any changed files opened with `CloudPath.open` to the cloud as soon as they are closed, it is safe to delete the cached version of the file at any point as long as the file is not opened for writing at the time you are trying to remove it. If necessary, the file will be re-downloaded next time it is needed.
+# 
+# We provide a number of ways to clear the file cache that can be useful if you want to reclaim disk space or if you don't need to keep cached files around.
+# 
+
+# 
+# ## Manually
+# 
+# It is recommended you pick an automatic strategy that works for your application. However, if you need to, you can clear the cache manually in three different ways: for individual CloudPath files, at the level of a `Client` instance, or at the operating system level. 
+# 
+#  - `CloudPath.clear_cache()` - for an individual `CloudPath` remove the cached version of the file if it exists.
+#  - `*Client.clear_cache()` - All files downloaded by this specific client instance will be removed from the cache. If you didn't create a client instance yourself, you can get the one that is used by a cloudpath with `CloudPath.client` or get the default one for a particular provider with `get_default_client`, for example by calling `S3Client.get_default_client().clear_cache()`.
+#  - By deleting the cached file itself or the containing directory using any normal method. To see where on a disk the cache is, you can use `CloudPath.fspath` for an individual file or use `*Client._local_cache_dir` for the client's cache. You can then use any method you like to delete these local files.
+# 
+# However, for most cases, you shouldn't need to manage the file cache manually. By setting the automatic cache clearing beahvior to the most appropriate one for your use case below, you can have the cache automatically cleared.
+# 
+
+# 
+# ## Automatically
```

Comment:
This is a general style preference that I don't hold super strongly but would apply throughout the rest of this notebook:

I think it may read more clearly to mention once that all of the options are defined on the `FileCacheMode` enum and then use the strings to refer to them, with quotes to be explicit that these are strings.

e.g., 

> These range from no cache clearing done by `cloudpathlib` (`"persistent"`), to the most aggressive (`"close_file"`)

> ## File cache mode: `"close_file"`